code/datums/components/possessable.dm

```diff
@@ -0,0 +1,18 @@
+/datum/component/ondemand_possessable
+	var/amount = 0
+
+/datum/component/ondemand_possessable/Initialize()
+	if(!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
```

Comment:
This isn't how you component. Make the ghost clicking and displaying happen from inside here